src/lib/comms/sol-network-impl-linux.c

```diff
@@ -580,13 +580,13 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
     return true;
 }
 
-SOL_API bool
+SOL_API int
 sol_network_link_up(uint16_t link_index)
 {
     return sol_network_link_set_status(link_index, IFF_UP, IFF_UP);
```

Comment:
@amandawrcoelho You should change the functions `sol_network_link_set_status()` return value type